server/src/main/java/org/opensearch/cluster/node/DiscoveryNodeRole.java

```diff
@@ -193,15 +202,53 @@ public final String toString() {
 
     /**
      * Represents the role for a master-eligible node.
+     * @deprecated As of 2.0, because promoting inclusive language, replaced by {@link #CLUSTER_MANAGER_ROLE}
      */
+    @Deprecated
     public static final DiscoveryNodeRole MASTER_ROLE = new DiscoveryNodeRole("master", "m") {
 
+        @Override
+        public Setting<Boolean> legacySetting() {
+            // copy the setting here so we can mark it private in org.opensearch.node.Node
+            // As of 2.0, set the default value to 'false', so that MASTER_ROLE isn't added as a default value of NODE_ROLES_SETTING
+            return Setting.boolSetting("node.master", false, Property.Deprecated, Property.NodeScope);
+        }
+
+    };
+
+    /**
+     * Represents the role for a cluster-manager-eligible node.
+     */
+    public static final DiscoveryNodeRole CLUSTER_MANAGER_ROLE = new DiscoveryNodeRole("cluster_manager", "m") {
+
         @Override
         public Setting<Boolean> legacySetting() {
             // copy the setting here so we can mark it private in org.opensearch.node.Node
             return Setting.boolSetting("node.master", true, Property.Deprecated, Property.NodeScope);
         }
 
+        @Override
+        public DiscoveryNodeRole getCompatibilityRole(Version nodeVersion) {
+            if (nodeVersion.onOrAfter(Version.V_2_0_0)) {
+                return this;
+            } else {
+                return DiscoveryNodeRole.MASTER_ROLE;
+            }
+        }
+
+        @Override
+        public void validateRole(List<DiscoveryNodeRole> roles) {
+            if (roles.contains(DiscoveryNodeRole.MASTER_ROLE)) {
+                throw new IllegalArgumentException(
+                    String.format(
+                        Locale.ROOT,
+                        "To promote inclusive language, [%s] role is deprecated, and is replaced by [%s] role. The two roles can not be assigned together to a node.",
+                        DiscoveryNodeRole.MASTER_ROLE.roleName(),
+                        DiscoveryNodeRole.CLUSTER_MANAGER_ROLE.roleName()
+                    )
+                );
+            }
+        }
```

Comment:
@andrross Thanks for your opinion to not allow "master" role and "cluster_manager" role coexist. 👍 I added a method validateRole() to allow adding custom validation rule for each node role, to check the compatibility with the other roles. The changes are in commit https://github.com/opensearch-project/OpenSearch/pull/2424/commits/0dadd6850b7c1200daec5c53a02291acc9a15541 https://github.com/opensearch-project/OpenSearch/pull/2424/commits/f818ea1c6c53a3d885847ad93e14c4d385a9014e, and unit test https://github.com/opensearch-project/OpenSearch/pull/2424/commits/5847ae0c5580512456d862fbbcf7dc850c06a127